packages/botonic-plugin-hubtype-analytics/src/event-models/ht-event.ts

```diff
@@ -0,0 +1,61 @@
+import { BaseEventData, EventName, HtEventProps, RequestData } from '../types'
+import { HtEventAgentRating } from './ht-event-agent-rating'
+import { HtEventBotAiModel } from './ht-event-bot-ai-model'
+import { HtEventBotFaq } from './ht-event-bot-faq'
+import { HtEventBotKeywordModel } from './ht-event-bot-keyword-model'
+import { HtEventBotRating } from './ht-event-bot-rating'
+import { HtEventChannelRating } from './ht-event-channel-rating'
+import { HtEventFaqUseful } from './ht-event-faq-useful'
+import { HtEventHandoffFail } from './ht-event-handoff-fail'
+import { HtEventHandoffSuccess } from './ht-event-handoff-success'
+
+export class HtEvent {
+  chat: string
+  event_type: EventName
+  event_data: BaseEventData
+
+  protected constructor(event: HtEventProps, requestData: RequestData) {
+    this.chat = requestData.userId
+    this.event_type = event.event_type
+    this.event_data = {
+      channel: requestData.provider,
+      event_datetime: new Date().toISOString(),
+      enduser_language: requestData.language,
+      enduser_country: requestData.country,
+      format_version: 1,
+      // flow_version?: string
```

Comment:
can this comment be removed?

Reply:
I have a doubt with this attribute, I don't know if I have to define it in the plugin or not. I will delete it in the future if I don't have to define it here.